File: smartmoneyserver/src/main/java/fpt/aptech/server/enums/plannedtransaction/RepeatType.java
package fpt.aptech.server.enums.plannedtransaction;

import lombok.Getter;

/**
 * Định nghĩa các kiểu lặp lại của giao dịch dự kiến.
 * Dùng cho tPlannedTransactions.repeat_type
 */
@Getter
public enum RepeatType {
    NONE(0),            // Không lặp lại
    DAILY(1),           // Hàng ngày
    WEEKLY(2),          // Hàng tuần
    MONTHLY(3),         // Hàng tháng
    YEARLY(4);          // Hàng năm

    private final int value;

    RepeatType(int value) {
        this.value = value;
    }
}

File: APP MOBILE/smart_money/lib/modules/planned/enums/repeat_type.dart
/// Kiểu lặp lại: Không, Hàng ngày, Tuần, Tháng, Năm
enum RepeatType {
  none,     // 0 - Không lặp
  daily,    // 1 - Hàng ngày
  weekly,   // 2 - Hàng tuần
  monthly,  // 3 - Hàng tháng
  yearly;   // 4 - Hàng năm

  int get value => index;

  static RepeatType fromValue(int value) {
    if (value < 0 || value > 4) {
      return RepeatType.none;
    }
    return RepeatType.values[value];
  }

  String get displayName {
    switch (this) {
      case RepeatType.none:
        return 'No repeat';
      case RepeatType.daily:
        return 'Daily';
      case RepeatType.weekly:
        return 'Weekly';
      case RepeatType.monthly:
        return 'Monthly';
      case RepeatType.yearly:
        return 'Yearly';
    }
  }
}

